bin/hdfcoinc/pycbc_multiifo_add_statmap

```diff
@@ -0,0 +1,214 @@
+#!/bin/env python
+""" Reduce IFAR for coincidences in times with more than one ifo combination
+available. Cluster for best IFAR. This clusters to find the most
+significant foreground, but leaves the background triggers alone.
+"""
+
+import h5py, numpy as np, argparse, logging, pycbc, pycbc.events, pycbc.io, lal
+import pycbc.version
+import pycbc.conversions as conv
+from pycbc.events import coinc
+from ligo import segments
+import itertools
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--version", action="version", version=pycbc.version.git_verbose_msg)
+parser.add_argument('--verbose', action='store_true')
+parser.add_argument('--statmap-files', nargs='+',
+                    help="List of coinc files to be redistributed")
+parser.add_argument('--censor-ifar-threshold', type=float, default=0.003,
+    help="If provided, only window out foreground triggers with IFAR (years)"
+         "above the threshold [default=0.003yr]")
+parser.add_argument('--veto-window', type=float, default=0.1,
+    help="Time around each zerolag trigger to window out [default=.1s]")
+parser.add_argument('--cluster-window', type=float)
+parser.add_argument('--output-file', help="name of output file")
+args = parser.parse_args()
+
+pycbc.init_logging(args.verbose)
+
+files = [h5py.File(n, 'r') for n in args.statmap_files]
+
+f = h5py.File(args.output_file, "w")
+
+logging.info('Copying segments and attributes to %s' % args.output_file)
+# Move segments information into the final file - remove some duplication
+# in earlier files. Also set up dictionaries to contain segments from the
+# individual statmap files
+indiv_segs = segments.segmentlistdict({})
+for fi in files:
+    key = fi.attrs['ifos'].replace(' ','')
+    starts = fi['segments/{}/start'.format(key)][:]
+    ends = fi['segments/{}/end'.format(key)][:]
+    indiv_segs[key] = pycbc.events.veto.start_end_to_segments(starts, ends)
+    f['segments/{}/start'.format(key)] = starts
+    f['segments/{}/end'.format(key)] = ends
+    if 'segments/foreground_veto' in fi:
+        f['segments/%s/foreground_veto/end' % key] = \
+                                         fi['segments/foreground_veto/end'][:]
+        f['segments/%s/foreground_veto/start' % key] = \
+                                       fi['segments/foreground_veto/start'][:]
+    for attr_name in fi.attrs:
+        if key not in f:
+            f.create_group(key)
+        f[key].attrs[attr_name] = fi.attrs[attr_name]
+
+logging.info('Combining foreground segments')
+
+# Convert segmentlistdict to a list ('seglists') of segmentlists
+# then np.sum(seglists, axis=0) does seglists[0] + seglists[1] + ...
+foreground_segs = np.sum(list(indiv_segs.values()), axis=0)
+# obtain list of all ifos involved in the coinc_statmap files
+all_ifos = np.unique([ifo for fi in files
+                      for ifo in fi.attrs['ifos'].split(' ')])
+
+# output to file
+f.attrs['foreground_time'] = abs(foreground_segs)
```

Comment:
this line belongs just after `foreground_segs` is calculated - also can remove the comment (it's obvious) and put in a blank line before the line that calculates 'all_ifos'